library/common/main_interface.cc

```diff
@@ -21,9 +22,15 @@
 
 static std::unique_ptr<Envoy::MainCommon> main_common_;
 static std::unique_ptr<Envoy::Http::Dispatcher> http_dispatcher_;
+static std::atomic<envoy_stream_t> current_stream_id_{0};
```

Comment:
I am assuming the vision is to make both this and the dispatcher owned by the engine instance once that work is done?

Reply:
Nope - stream handles will be necessarily globally unique. This is so that we can locate the appropriate the engine/dispatcher by just the handle.

Reply:
just realized I didn't update the name here though, let me fix

Reply:
And hence why init_stream takes an envoy_engine_t gotcha.